101-machine-learning-create/azuredeploy.json

```diff
@@ -1,132 +1,135 @@
 {
-  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-  "contentVersion": "1.0.0.0",
-  "parameters": {
-    "workspaceName": {
-      "type": "string",
-      "metadata": {
-        "description": "Specifies the name of the Azure Machine Learning service workspace."
-      }
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+            "defaultValue": "[concat('WorkSpace-', uniqueString(resourceGroup().id))]",
+            "metadata": {
+                "description": "Specifies the name of the Azure Machine Learning service workspace."
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "eastus",
```

Comment:
remove defaultValue

Reply:
Perhaps treat the location from the resource group is a better practice?
Then I will specify about the location during the Resource group creation and let the defaultValue getting the location from the RG.